src/scancode/resource.py

```diff
@@ -434,19 +436,40 @@ def create_resources(_seq, _top, _parent, _is_file):
         # not keep parents already walked and we walk topdown.
         parent_by_loc = {root.location: root}
 
+
+        def _walk():
+            """Walk through the root_directory recursively upto max_depth nested subdirectories
+            and build the Resource tree. max_depth should be positive or -1 for walking through
+            maximum available nesting levels."""
+
+            # Find root directory depth using path separator's count
+            root_dir_depth = root.location.count(os.path.sep)
+
+            # Default behaviour. current_depth is always less than max_depth so that 
+            # os_walk keeps running. When max_depth is limited, the current_depth is
+            # changed in each iteration and compared with max_depth
+            current_depth = -2
+
+            for top, dirs, files in os_walk(root.location, topdown=True, onerror=err):
+                # If depth is limited
+                if self.max_depth >= 0:
```

Comment:
`self.max_depth` is accessed twice in the loop may be it would make sense to define `max_depth = self.max_depth` outside of the loop? and then use it?

Reply:
Yeah. Thanks.